spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -1204,9 +1226,24 @@ def process_symbols(self, params):
             self.log_lsp_handle_errors("Error when processing symbols")
 
     # ------------- LSP: Linting ---------------------------------------
-    def schedule_document_did_change(self):
+    def _schedule_document_did_change(self):
         """Schedule a document update."""
         self._document_server_needs_update = True
+
+        # If auto-completions are on, start the timer to request completions
+        # here. That way we can ensure an almost instant response when the user
+        # stops typing.
+        # Note: We only ask for auto-completions if there was some text
+        # generated as part of the keyPressEvent.
+        # Fixes spyder-ide/spyder#11021
+        if (
+            self.automatic_completions
+            and not self._server_requests_timer.isActive()
```

Comment:
Why are you checking `self._server_requests_timer.isActive()`?

Reply:
The idea is to use this timer to limit the amount of completion requests send to the server. So, we send a request when the starts to type and prevent to do it while it's still doing it.

I think it has the role you thought `automatic_completions_after_ms` had.

Reply:
I am not sure about that. If `_server_requests_timer` is active, the automatic completion would never trigger?
I would just drop this test. If `automatic_completions_after_ms` is long enough that the user has time to type a new key, `start()` will interrupt the timer and restart it, preventing the previous completion request from being sent.